java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/RuntimeImageBuildStep.java

```diff
@@ -72,27 +74,27 @@ private String getBaseRuntimeImage(BuildContext buildContext, Artifact artifact)
     RuntimeConfig runtimeConfig = buildContext.getRuntimeConfig();
 
     // Check if the user has explicitly selected the compat runtime
-    if (buildContext.isCompatEnabled() && artifact.getType() != EXPLODED_WAR) {
-      throw new BuildStepException(String.format("App Engine APIs have been enabled. In order to "
-          + "use App Engine APIs, an exploded WAR artifact is required, but a %s artifact was "
-          + "found. See https://cloud.google.com/appengine/docs/flexible/java/upgrading for more "
-          + "detail.", artifact.getType()));
+    if (buildContext.isCompatEnabled()) {
+      if (artifact.getType() != EXPLODED_WAR) {
+        throw new BuildStepException(String.format("App Engine APIs have been enabled. In order to "
+            + "use App Engine APIs, an exploded WAR artifact is required, but a %s artifact was "
+            + "found. See https://cloud.google.com/appengine/docs/flexible/java/upgrading for more "
+            + "detail.", artifact.getType()));
+      }
+      return compatImageName;
```

Comment:
Can we refactor to just have one return statement at the end of the method?
Returning here will miss the logging statement, and we don't want to do that.
```
 logger.info("Using base image '{}' for {} artifact", baseImage, artifactType);
```

Reply:
Will do.